### PR TITLE
Change mpifx_comm to mpi_f08

### PR DESCRIPTION
**Pull request by jubich**

### CODE REVIEW

File: lib/mpifx_comm.fpp [hunk 1]
```diff
@@ -102,14 +125,15 @@ contains
     class(mpifx_comm), intent(out) :: newcomm
     integer, intent(out), optional :: error
 
-    integer :: error0, newcommid
+    integer :: error0
+    type(mpi_comm) :: newcomm_
```
Comment (aradi) [inline lib/mpifx_comm.fpp:129]: Let's use `newmpicomm` instead of the `newcomm_` with the trailing underscore.

**Review by aradi — CHANGES_REQUESTED** [review]

File: lib/mpifx_comm.fpp [hunk 2]
```diff
@@ -38,28 +43,46 @@ contains
...
       return
     end if
     self%leadrank = 0
     self%lead = (self%rank == self%leadrank)
 
-  end subroutine mpifx_comm_init
+  end subroutine mpifx_comm_init_comm
+
+
+  subroutine mpifx_comm_init_int(self, commid, error)
```
Comment (bhourahine) [inline lib/mpifx_comm.fpp:75]: Can you please add comments for the routine, but also why is is called _init_int() ? This is not very informative.
```suggestion
  !> Initializes a communicator from a numerical id.
  !!
  !! \param self  Initialized instance on exit.
  !! \param commid  Numerical MPI Communicator ID
  !! \param error  Error flag on return containing the first error occurring
  !!     during the calls mpi_comm_size and mpi_comm_rank.
  !!
  subroutine mpifx_comm_from_id(self, commid, error)
```

File: lib/mpifx_comm.fpp [hunk 3]
```diff
@@ -38,28 +43,46 @@ contains
...
+    class(mpifx_comm), intent(out) :: self
+    integer, intent(in) :: commid
+    integer, intent(out), optional :: error
+
+    type(mpi_comm) :: newcomm
+
+    newcomm%mpi_val = commid
+    call self%mpifx_comm_init_comm(newcomm, error)
+
+  end subroutine mpifx_comm_init_int
```
Comment (bhourahine) [inline lib/mpifx_comm.fpp:85]:
```suggestion
  end subroutine mpifx_comm_from_id
```

File: lib/mpifx_comm.fpp [hunk 4]
```diff
@@ -9,14 +9,19 @@ module mpifx_comm_module
...
+    type(mpi_comm) :: comm  !< MPI communicator handle.
+    integer :: size         !< Nr. of processes (size).
+    integer :: rank         !< Rank of the current process.
+    integer :: leadrank     !< Index of the lead node.
+    logical :: lead         !< True if current process is the lead (rank == 0).
   contains
+
     !> Initializes the MPI environment.
-    procedure :: init => mpifx_comm_init
+    procedure, private :: mpifx_comm_init_int
```
Comment (bhourahine) [inline lib/mpifx_comm.fpp:21]:
```suggestion
    procedure, private :: mpifx_comm_from_id
```

File: lib/mpifx_comm.fpp [hunk 5]
```diff
@@ -9,14 +9,19 @@ module mpifx_comm_module
...
+    procedure, private :: mpifx_comm_init_comm
```
Comment (bhourahine) [inline lib/mpifx_comm.fpp:22]:
```suggestion
    procedure, private :: mpifx_comm_from_type
```

File: lib/mpifx_comm.fpp [hunk 6]
```diff
@@ -9,14 +9,19 @@ module mpifx_comm_module
...
+
+    generic :: init => mpifx_comm_init_int, mpifx_comm_init_comm
```
Comment (bhourahine) [inline lib/mpifx_comm.fpp:24]:
```suggestion
    generic :: init => mpifx_comm_from_id, mpifx_comm_from_type
```

File: lib/mpifx_comm.fpp [hunk 7]
```diff
@@ -38,28 +43,46 @@ contains
   !! \param error  Error flag on return containing the first error occuring
   !!     during the calls mpi_comm_size and mpi_comm_rank.
   !!
-  subroutine mpifx_comm_init(self, commid, error)
+  subroutine mpifx_comm_init_comm(self, comm, error)
```
Comment (bhourahine) [inline lib/mpifx_comm.fpp:46]:
```suggestion
  subroutine mpifx_comm_from_type(self, comm, error)
```

File: lib/mpifx_comm.fpp [hunk 8]
```diff
@@ -38,28 +43,46 @@ contains
...
+    call mpi_comm_rank(self%comm, self%rank, error0)
     call handle_errorflag(error0, "mpi_comm_rank() in mpifx_comm_init()", error)
     if (error0 /= 0) then
       return
     end if
     self%leadrank = 0
     self%lead = (self%rank == self%leadrank)
 
-  end subroutine mpifx_comm_init
+  end subroutine mpifx_comm_init_comm
```
Comment (bhourahine) [inline lib/mpifx_comm.fpp:72]:
```suggestion
  end subroutine mpifx_comm_from_type
```

File: lib/mpifx_comm.fpp [hunk 9]
```diff
@@ -38,28 +43,46 @@ contains
   !! \param error  Error flag on return containing the first error occuring
```
Comment (bhourahine) [inline lib/mpifx_comm.fpp:43]:
```suggestion
  !! \param error  Error flag on return containing the first error occurring
```

**Review by bhourahine — COMMENTED** [review]
I can't make a suggestion for line 42 (due to github's limitation to only close to changed lines):
\param commid  MPI Communicator (default: \c MPI_COMM_WORLD)

**Review by bhourahine — APPROVED** [review]
Ready to merge

**Pull request merged**